metagraph/integration_tests/test_assemble.py

```diff
@@ -195,3 +200,132 @@ def test_round_robin_graph_size_via_gfa(self, gfa_test):
+                      'row_diff_brwt_no_anchor_opt']
+    ),
+    class_name_func=get_test_class_name
+)
+class TestDiffAssembly(TestingBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.tempdir = TemporaryDirectory()
+
+        cls.with_bloom = False
+        if cls.graph_repr == 'succinct_bloom':
+            cls.graph_repr = 'succinct'
+            cls.with_bloom = True
+
+        cls.mask_dummy = False
+        if cls.graph_repr == 'succinct_mask':
+            cls.graph_repr = 'succinct'
+            cls.mask_dummy = True
+
+        k = 31
+        construct_command = '{exe} build {mask_dummy} -p {num_threads} \
+                --graph {repr} -k {k} -o {outfile} {input}'.format(
+            exe=METAGRAPH,
+            num_threads=NUM_THREADS,
+            mask_dummy='--mask-dummy' if cls.mask_dummy else '',
+            repr=cls.graph_repr,
+            k=k,
+            outfile=cls.tempdir.name + '/graph',
+            input=TEST_DATA_DIR + '/metasub_fake_data_simple.fa'
+        )
+        res = subprocess.run([construct_command], shell=True)
+        assert(res.returncode == 0)
+
+        if cls.with_bloom:
+            convert_command = '{exe} transform -o {outfile} --initialize-bloom {bloom_param} {input}'.format(
+                exe=METAGRAPH,
+                outfile=cls.tempdir.name + '/graph',
+                bloom_param='--bloom-fpp 0.1',
+                input=cls.tempdir.name + '/graph' + graph_file_extension[cls.graph_repr],
+            )
+            res = subprocess.run([convert_command], shell=True)
+            assert(res.returncode == 0)
+
+        def check_suffix(anno_repr, suffix):
+            match = anno_repr.endswith(suffix)
+            if match:
+                anno_repr = anno_repr[:-len(suffix)]
+            return anno_repr, match
+
+        cls.anno_repr, separate = check_suffix(cls.anno_repr, '_separate')
+        cls.anno_repr, no_fork_opt = check_suffix(cls.anno_repr, '_no_fork_opt')
+        cls.anno_repr, no_anchor_opt = check_suffix(cls.anno_repr, '_no_anchor_opt')
+
+        cls._annotate_graph(
+            TEST_DATA_DIR + '/metasub_fake_data_simple.fa',
+            cls.tempdir.name + '/graph' + graph_file_extension[cls.graph_repr],
+            cls.tempdir.name + '/annotation',
+            cls.anno_repr,
+            separate,
+            no_fork_opt,
+            no_anchor_opt
+        )
+
+    def test_diff_assembly(self):
+        assemble_command = '{exe} assemble -p {num_threads} \
+                -a {annotation} -o {outfile} \
+                --diff-assembly-rules {mask} {graph}'.format(
+            exe=METAGRAPH,
+            num_threads=NUM_THREADS,
+            outfile=self.tempdir.name + '/diff_contigs',
+            graph=self.tempdir.name + '/graph' + graph_file_extension[self.graph_repr],
+            annotation=self.tempdir.name + '/annotation' + anno_file_extension[self.anno_repr],
+            mask=TEST_DATA_DIR + '/example.diff.json'
+        )
```

Comment:
f-strings don't work as well here since the variable values are not precomputed. If I did it that way, I'd have to define a bunch of these variables in the lines above, so it wouldn't look as nice

Reply:
what do you mean? You don't need to precompute anything

Reply:
```suggestion
        assemble_command = f'{METAGRAPH} assemble -p {NUM_THREADS} \
                -a {self.tempdir.name}/annotation{anno_file_extension[self.anno_repr]} \
                -o {self.tempdir.name}/diff_contigs \
                --diff-assembly-rules {TEST_DATA_DIR}/example.diff.json \
                {self.tempdir.name}/graph{graph_file_extension[self.graph_repr]}'
```

Reply:
ok, I tried it out like this just now and it worked. I looked back at what I was doing before and for some reason, I tried stuff like this and it was complaining...
```python
{self.tempdir.name + "diff_contigs"}
```

but yes, I should have done it as suggested

Reply:
this thing works as well, you just need to escape the quotes correctly